app/javascript/react/src/components/Greeting.jsx

```diff
@@ -0,0 +1,30 @@
+import React, { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { useState } from 'react';
+import fetchData from "../redux/welcome/fetchAPi";
+
+const RandomMessage = () => {
+  const [msg, setMsg] = useState('');
+  const state = useSelector((state) => state.Random);
+  const dispatch = useDispatch();
+  
+  useEffect(() => {
+    dispatch(fetchData());
+  }, [dispatch]);
+
+  useEffect(() => {
+    if (state.data) {
+      setMsg(state.data.message);
+    }
+  }, [state.data]);
```

Comment:
**OPTIONAL** - You have done well so far, but this is not an efficient way to code, you are dong the same thing twice, hence you are not following the YAGNI principle, since you are using Redux to manage your app's state, you do not need to use react `useState` hook to set the state again. You can directly pass the app's state from your redux store into your component. Please fix this 🙏